tests/test_all_component_and_model_builder.py

```diff
@@ -0,0 +1,296 @@
+"""Test model builder using variance of parfile input"""
+
+from collections import defaultdict
+import pytest
+import io
+import astropy.units as u
+from pint.models.timing_model import (
+    PhaseComponent,
+    Component,
+    AllComponents,
+    ConflictAliasError,
+    UnknownBinaryModel,
+)
+from pint.models.model_builder import ModelBuilder, ComponentConflict
+from pint.models.parameter import floatParameter
+from pint.utils import split_prefixed_name, PrefixError
+
+
+class SimpleModel(PhaseComponent):
+    """Very simple test model component
+    """
+
+    register = True
+    category = "simple_test"
+
+    def __init__(self):
+        super(SimpleModel, self).__init__()
+        self.add_param(floatParameter(name="TESTPARAM", value=0.0, unit="s"))
+
+
+class SubsetModel(PhaseComponent):
+    """Test model component hosting the parameters which are a subset of spindown.
+    """
+
+    register = False  # This has to be false, otherwrise all test will fail.
+    category = "simple_test"
+
+    def __init__(self):
+        super(SubsetModel, self).__init__()
+        self.add_param(floatParameter(name="F0", value=0.0, unit="1/s"))
+        self.add_param(floatParameter(name="F1", value=0.0, unit="1/s^2"))
+
+
+@pytest.fixture
+def sub_set_model():
+    return SubsetModel()
+
+
+@pytest.fixture
+def simple_model():
+    return SimpleModel()
+
+
+@pytest.fixture
+def simple_model_overlap(simple_model):
+    simple_model.add_param(floatParameter(name="F0", aliases=[], value=0.0, unit="1/s"))
+    return simple_model
+
+
+@pytest.fixture
+def simple_model_alias_overlap():
+    simple_model = SimpleModel()
+    simple_model.add_param(
```

Comment:
Shouldn't this raise an exception? Trying to add any parameter that shares a name or alias with an existing parameter?

Reply:
That is a good point. This will have to modify the add_parameter code, which is quite critical. So I think it is better to do it in another PR.